python/delete-default-vpcs.py

```diff
@@ -1,84 +1,53 @@
 #!/usr/bin/env python
-#activate virtual environment, e.g.: source /Users/Paul/venv/python3/bin/activate
-#for testing: python delete-default-vpcs.py --profile management --dry_run
-import boto3
-import sys
-import time
-import json
+# activate virtual environment, e.g.: source /Users/Paul/venv/python3/bin/activate
+# for testing: python delete-default-vpcs.py --profile management --dry_run
 import argparse
-from sessions import get_session
+import time
 
+import boto3
 from botocore.exceptions import ClientError
-from datetime import datetime
-
-def json_serial(obj):
-    """JSON serializer for objects not serializable by default json code"""
-
-    if isinstance(obj, datetime):
-        serial = obj.isoformat()
-        return serial
-    raise TypeError("Type not serializable")
-
-class iam:
-    def __init__(self, profile, account_number, stage, deployment_type, dry_run, external_id):
-
-        self.profile = profile
-        self.account_number = account_number
-        self.dry_run = dry_run
-
-        if deployment_type == "github":
-            self.session = get_session(account_number, "TDRTerraformRole" + stage.capitalize(), external_id)
-        else:
-            self.session = boto3.session.Session(profile_name=self.profile)
-
-        self.client = self.session.client('iam')
 
-        aliases = self.client.list_account_aliases()['AccountAliases']
-        #print(json.dumps(aliases, sort_keys=True, indent=2, default=json_serial))
+from sessions import get_session
 
-        print('Deleting VPCs in Account %s' % aliases[0])
 
 class ec2:
-    def __init__(self, profile, account_number, stage, deployment_type, dry_run, external_id):
+    def __init__(self, profile, account_number, deployment_type, dry_run, external_id, role_name):
 
         self.profile = profile
         self.account_number = account_number
         self.dry_run = dry_run
 
         if deployment_type == "github":
-            self.session = get_session(account_number, "TDRTerraformRole" + stage.capitalize(), external_id)
+            self.session = get_session(account_number, role_name, external_id)
         else:
             self.session = boto3.session.Session(profile_name=self.profile)
 
-        self.client = self.session.client('ec2')
+        self.client = self.session.client('ec2', region_name="eu-west-2")
```

Comment:
There's a couple of places in python scripts where the `region_name` needs to be set, is it possible to set it in some python config?

Reply:
Fair enough, can just leave as is.